### PR TITLE
[@entria/relay-experimental] Create new package

### PR DESCRIPTION
**Pull request by renanmav**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by renanmav** [comment]
@alloy I removed `../..` imports that I missed in #38640

**Comment by RyanCavanaugh** [comment]
@renanmav looks good to go from my side, shall I merge or do you want to wait for alloy?

**Comment by renanmav** [comment]
@RyanCavanaugh my changes on `relay-runtime` are not significant. We can merge this.

**Comment by renanmav** [comment]
Sorry, I forgot that my master branch was synced with this PR

**Comment by alloy** [comment]
Nice 👌 

It might be helpful to add a comment to the non-experimental packages that mentions to look at this package when in the future some of these APIs move out of the experimental phase? Nothing urgent, though.

**Comment by renanmav** [comment]
> Nice 👌
> 
> It might be helpful to add a comment to the non-experimental packages that mentions to look at this package when in the future some of these APIs move out of the experimental phase? Nothing urgent, though.

Any example of non-experimental packages? @alloy

**Comment by alloy** [comment]
Yeah I meant `@types/relay-runtime`, `@types/react-relay`, `@types/relay-compiler`

**Comment by renanmav** [comment]
We should embed definitions on source code of these packages, instead install definitions through `@types`. What do you think?

**Comment by alloy** [comment]
I’m unsure what you mean, could you rephrase?

**Comment by renanmav** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package

Could we bundle the declaration files in relay source code?

Something like the `typescript` folder of [`react-navigation`](https://github.com/react-navigation/react-navigation)

**Comment by alloy** [comment]
Ahh, I don’t see that happening any time soon and it would also probably slow down the process of iteration.
